### PR TITLE
Template sets squareup

### PR DESCRIPTION
**Pull request by moljac**
### Describe your contribution

1. Moved `Square.*` to this repo from:
   https://github.com/xamarin/XamarinComponents/tree/main/Android/SquareOkHttp
   https://github.com/xamarin/XamarinComponents/tree/main/Android/SquareOkHttp3
   https://github.com/xamarin/XamarinComponents/tree/main/Android/SquareOkHttp3UrlConnection
   https://github.com/xamarin/XamarinComponents/tree/main/Android/SquareOkIO
   https://github.com/xamarin/XamarinComponents/tree/main/Android/SquarePicasso
   https://github.com/xamarin/XamarinComponents/tree/main/Android/SquareRetrofit
2. created template sets

### Does this change any of the generated binding API's?

`net6-android` TFM added

Checklist
- [x] Add any metadata needed to get PR to compile
- [x] Verify any new namespaces are properly renamed (utilities.cake)
- [x] Verify nuget/namespace spelling (utilities.cake)           

SquareUp dependencies moved to this repo as template sets

### CODE REVIEW

File: source/GooglePlayServicesProject.cshtml [hunk 1]
```diff
@@ -143,19 +143,34 @@
       }
   </ItemGroup>
 
+  <!--
```
Comment (jpobst) [inline source/GooglePlayServicesProject.cshtml:146]: Probably don't need to commit this file.

**Review by jpobst — COMMENTED** [review]
Needs some changes to build on CI.

**Comment by moljac** [comment]
New on CI. Issue needs to be opened (`samples/Nuget.config`)

```
For more information please visit: https://aka.ms/nugetsecurityanalysis
Starting Multifeed Nuget Security Analysis: 
Starting Multifeed Python Security Analysis: 
Starting CFS NuGet Analysis: 
samples/NuGet.Config - External feeds found in the file.
Starting CFS NPM Analysis: 
##[warning]Azure Artifacts Configuration Analysis found 1 vulnerable NuGet package manifest in the repository. Visit http://aka.ms/azure-artifacts-configuration-analysis for more details.
Finishing: Nuget Security Analysis (auto-injected by policy)
```

File: source/GooglePlayServicesProject.cshtml (hunk 1 above)
Comment (moljac) [reply source/GooglePlayServicesProject.cshtml:146]: `JavaSourceJar`and `JavaDocJar` parts are commented out, but `if` contains artifact list of the artifacts that cause `java-source-utils.jar`to fail. Thus I added it to the PR. I will work on issue as soon as .NET6 migration and template sets are done.

**Comment by moljac** [comment]
https://repo1.maven.org/maven2/com/squareup/okhttp/okhttp/2.7.5/okhttp-2.7.5.pom

https://repo1.maven.org/maven2/com/squareup/okhttp/parent/2.7.5/parent-2.7.5.pom

okio 1.6.0

https://www.nuget.org/packages/Square.OkIO/1.6.0

https://repo1.maven.org/maven2/com/squareup/okio/okio/

trying

https://repo1.maven.org/maven2/com/squareup/okio/okio/2.10.0/okio-2.10.0.pom

**Comment by moljac** [comment]
retrofit pom.xml:

https://repo1.maven.org/maven2/com/squareup/retrofit/retrofit/1.9.0/retrofit-1.9.0.pom

File: source/GooglePlayServicesProject.cshtml [hunk 2]
```diff
@@ -268,15 +283,19 @@
...
     {
-      <PackageReference Include="Xamarin.Google.Guava" Version="28.2.0.1" />
+      <PackageReference Include="Xamarin.Google.Guava" Version="29.0.0" />
     }
     @if (@Model.NuGetPackageId == "Xamarin.Firebase.Crashlytics")
     {
-      <PackageReference Include="Xamarin.Google.Dagger" Version="2.27.0" />
+      <PackageReference Include="Xamarin.Google.Dagger" Version="2.37.0" />
+    }
+    @if (@Model.NuGetPackageId == "Xamarin.Firebase.InAppMessaging")
```
Comment (jpobst) [inline source/GooglePlayServicesProject.cshtml:296]: Let's remove this whole file from the PR.  Adding unrelated changes (even commented out) to PR's causes unnecessary review time and unnecessary merge conflicts.

In this case, this new `<PackageReference>` is not commented out and causes additional diffs and questions of why we would want to add this instead of using the automatic dependencies.

**Review by jpobst — COMMENTED** [review]
Let's get `GooglePlayServicesProject.cshtml` removed from this PR, then it looks good.

**Comment by jpobst** [comment]
Note: the `excludedRuntimeDependencies` are `test` dependencies specified on a parent POM, which we do not currently support.  We are using `excludedRuntimeDependencies` to get around this. ([Details](https://github.com/xamarin/XamarinComponents/pull/1291))

File: config.json [hunk 3]
```diff
@@ -1651,40 +1651,55 @@
...
+        "nugetId": "Square.Retrofit",
+        "dependencyOnly": false,
+        "templateSet": "squareup-retrofit",
+        "excludedRuntimeDependencies": "com.squareup.okhttp.okhttp,com.google.code.gson.gson,com.google.android.android,io.reactivex.rxjava,com.google.appengine.appengine-api-1.0-sdk"
+      },
+      {
+        "groupId": "com.squareup.okio",
+        "artifactId": "okio",
+        "version": "2.7.0",
+        "nugetVersion": "2.7.0",
```
Comment (jpobst) [inline config.json:1672]: The current NuGet version of this is `2.10.0`.

File: config.json [hunk 4]
```diff
@@ -1651,40 +1651,55 @@
...
+        "nugetId": "Square.OkIO",
+        "dependencyOnly": false,
+        "templateSet": "squareup-okio"
       },
       {
         "groupId": "com.squareup.okhttp3",
         "artifactId": "okhttp",
         "version": "4.8.1",
         "nugetVersion": "4.8.1",
```
Comment (jpobst) [inline config.json:1681]: The current NuGet version of this is `4.9.2`.

File: config.json [hunk 5]
```diff
@@ -1651,40 +1651,55 @@
...
+        "templateSet": "squareup-okhttp3"
       },
       {
         "groupId": "com.squareup.okhttp3",
         "artifactId": "okhttp-urlconnection",
         "version": "4.3.1",
-        "nugetVersion": "4.3.1",
-        "nugetId": "Square.OkHttp3",
-        "dependencyOnly": true
+        "nugetVersion": "4.3.1.1",
```
Comment (jpobst) [inline config.json:1690]: The current NuGet version of this is `4.9.2`.

**Review by jpobst — COMMENTED** [review]
Need to update a few versions so they match what is currently on NuGet, so the diffs don't show a lot of removed API.

**Review by jpobst — APPROVED** [review]
I bumped one more version to match NuGet, and added the needed dependency to get it to compile.  Feel free to double-check it.

LGTM

**Comment by moljac** [comment]
> I bumped one more version to match NuGet, and added the needed dependency to get it to compile. Feel free to double-check it.
> 

Thanks a lot

I did check. Mostly assembly versions in ApiDiff

> LGTM

merging
